src/main/java/uk/gov/justice/digital/client/glue/GlueSchemaClient.java

```diff
@@ -0,0 +1,52 @@
+package uk.gov.justice.digital.client.glue;
+
+import com.amazonaws.services.glue.AWSGlue;
+import com.amazonaws.services.glue.model.EntityNotFoundException;
+import com.amazonaws.services.glue.model.GetSchemaVersionRequest;
+import com.amazonaws.services.glue.model.SchemaId;
+import com.amazonaws.services.glue.model.SchemaVersionNumber;
+import jakarta.inject.Inject;
+import jakarta.inject.Singleton;
+import lombok.val;
+import uk.gov.justice.digital.config.JobArguments;
+
+import java.util.Optional;
+
+@Singleton
+public class GlueSchemaClient {
+
+    private final AWSGlue glueClient;
+    private final String contractRegistryName;
+
+    @Inject
+    public GlueSchemaClient(GlueClientProvider glueClientProvider,
+                            JobArguments jobArguments) {
+        this.glueClient = glueClientProvider.getClient();
+        this.contractRegistryName = jobArguments.getContractRegistryName();
+    }
+
+    public Optional<String> getSchema(String schemaName) {
+        try {
+            val result = glueClient.getSchemaVersion(createRequestForSchemaName(schemaName));
+            val schemaData = result.getSchemaDefinition();
+            return Optional.of(schemaData);
+        }
+        catch (EntityNotFoundException e) {
+            return Optional.empty();
+        }
+    }
+
+    private GetSchemaVersionRequest createRequestForSchemaName(String schemaName) {
```

Comment:
make the modifier as final in the parameter schemaName

Reply:
We agreed that final parameters don't really add anything.

Reply:
not necessarily true. The method should not alter the original data it sent. If you doesn't make it final, the method can alter the value and the calling place may have the corrupted value

Reply:
This is extremely unlikely and when one considers the cost / benefit it's not really worth the added visual noise.